### PR TITLE
RIA-3978 user would like to manipulate git branches using the git plugin

### PR DESCRIPTION
**Pull request by AlexanderNikitin-Smartbear**
Depends on:
https://github.com/SmartBear/readyapi-free-os/pull/220
https://github.com/SmartBear/axm/pull/8825

https://smartbear.atlassian.net/browse/RIA-3978

### CODE REVIEW

File: src/main/java/com/smartbear/readyapi/plugin/git/GitCommandHelper.java [hunk 1]
```diff
@@ -375,4 +380,57 @@ private boolean canMerge(final Git git, final MergeStrategy mergeStrategy) {
...
+                    .map(Ref::getName)
+                    .collect(Collectors.toList());
+        } catch (GitAPIException e) {
+            throw new VcsIntegrationException("Can not get list of branches", e);
+        }
+    }
+
+    public void checkout(String commitOrBrunch, final Git git) {
+        if (commitOrBrunch == null) {
+            throw new NullPointerException("commitOrBrunch is null");
```
Comment (Alexander-Paeshin) [inline src/main/java/com/smartbear/readyapi/plugin/git/GitCommandHelper.java]: I think, the IllegalArgumentException is more applicable in this case.

File: src/main/java/com/smartbear/readyapi/plugin/git/GitCommandHelper.java [hunk 2]
```diff
@@ -375,4 +380,57 @@ private boolean canMerge(final Git git, final MergeStrategy mergeStrategy) {
...
+        }
+        try {
+            if (!git.getRepository().getRepositoryState().canCheckout()) {
+                throw new VcsIntegrationException("Can not checkout!");
```
Comment (Alexander-Paeshin) [inline src/main/java/com/smartbear/readyapi/plugin/git/GitCommandHelper.java]: This error message looks like very strange. It is not informative. I suggest to check all messages with technical writers.

File: src/main/java/com/smartbear/readyapi/plugin/git/GitCommandHelper.java [hunk 3]
```diff
@@ -375,4 +380,57 @@ private boolean canMerge(final Git git, final MergeStrategy mergeStrategy) {
 
         return true;
     }
+
+    public String getCurrentBranch(final Git git) {
+        try {
+            return git.getRepository().getFullBranch();
+        } catch (IOException e) {
+            throw new VcsIntegrationException("Can get current branch", e);
```
Comment (Alexander-Paeshin) [inline src/main/java/com/smartbear/readyapi/plugin/git/GitCommandHelper.java]: As I see this error message is incorrect.

**Review by Alexander-Paeshin — CHANGES_REQUESTED** [review]

File: src/main/java/com/smartbear/readyapi/plugin/git/GitCommandHelper.java (hunk 3 above)
Comment (AlexanderNikitin-Smartbear) [reply src/main/java/com/smartbear/readyapi/plugin/git/GitCommandHelper.java]: Fixed

File: src/main/java/com/smartbear/readyapi/plugin/git/GitCommandHelper.java (hunk 1 above)
Comment (AlexanderNikitin-Smartbear) [reply src/main/java/com/smartbear/readyapi/plugin/git/GitCommandHelper.java]: Replaced

File: src/main/java/com/smartbear/readyapi/plugin/git/GitCommandHelper.java (hunk 2 above)
Comment (AlexanderNikitin-Smartbear) [reply src/main/java/com/smartbear/readyapi/plugin/git/GitCommandHelper.java]: I've added information to the message.

**Review by Alexander-Paeshin — APPROVED** [review]

**Pull request merged**